### ISSUE TITLE
Batch size setting

### ISSUE THREAD

**Issue by Eryo-iPython** [RUCKBReasoning/OmniSQL#44]
The batch size in the paper is set to 512 on 8 a800 GPUs. But in this code it is set to 2 per GPUs.  Setting the batch size to 512 results in memory overflow. I'd like to reproduce the experimental results. Would setting the batch size to 2 as suggested on GitHub work?

**Comment by lihaoyang-ruc** [comment]
In DeepSpeed, the effective batch size per training step is calculated as:

```
total_batch_size = PER_DEVICE_TRAIN_BATCH_SIZE × num_processes × gradient_accumulation_steps
```

Here, `num_processes` typically refers to the number of GPUs. Utilizing `gradient_accumulation_steps` is an effective strategy for addressing memory overflow: it accumulates gradients across several forward passes before performing a backward pass and optimizer update, allowing the simulation of a larger batch size without exceeding GPU memory limits.

Both `num_processes` and `gradient_accumulation_steps` can be configured in the accelerator’s config file (e.g., `accelerate_config_14b.yaml`, `accelerate_config_7b.yaml`, or `accelerate_config_32b.yaml`).

For example, if you set `PER_DEVICE_TRAIN_BATCH_SIZE` to 2, `num_processes` to 8, and `gradient_accumulation_steps` to 64, the effective batch size will be:

```
2 × 8 × 64 = 1024
```

You can adjust these values to best fit your hardware setup.

By the way, for a fixed total batch size and a fixed number of GPUs, I recommend increasing `PER_DEVICE_TRAIN_BATCH_SIZE` as much as your hardware allows, while decreasing `gradient_accumulation_steps`. This maximizes GPU utilization and can result in a faster training process.
